docs/application/native/guides/connectivity/softap.md

```diff
@@ -0,0 +1,341 @@
+
+   ```
+   <privileges>
+     <!--Needed for configuring SoftAP settings-->
+     <privilege>http://tizen.org/privilege/softap</privilege>
+     <!--Needed for enabling and disabling SoftAP-->
+     <privilege>http://tizen.org/privilege/softap.admin</privilege>
+   </privileges>
+   ```
+2. To use the functions and data types of the SoftAP API, include the `<softap.h>` header file in your application:
+
+   ```
+   #include <softap.h>
+   ```
+
+
+## Creating SoftAP handle
+
+To create SoftAP handle:
+
+   ```
+   softap_h softap;
+   int ret = SOFTAP_ERROR_NONE;
+
+   ret = softap_create(&softap);
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("softap_create failed: %d", ret);
+   ```
+
+
+
+
+## Configuring SoftAP settings
+
+To configure SoftAP settings:
+
+1. Set SSID with the `softap_set_ssid()` function. If you don't use this function, device nam is used as SSID:
+   ```
+   int ret = SOFTAP_ERROR_NONE;
+   softap_h softap = NULL;
+
+   ret = softap_set_ssid(softap, "Tizen");
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Failed to set SSID");
+   ```
+
+2. Set security type and passphrase. If the passphrase is not set, random string of 8 characters is used:
+   ```
+   ret = softap_set_security_type(softap, SOFTAP_SECURITY_TYPE_WPA2_PSK);
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Failed to set security type");
+
+   ret = softap_set_passphrase(softap, "1q2w3e4r!@");
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Failed to set passphrase");
+   ```
+
+3. Set visibility of SSID. In this example, client can find SoftAP device from Wi-Fi scan list.
+   ```
+   ret = softap_set_ssid_visibility(softap, true);
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Failed to set visibility");
+   ```
+
+## Enabling SoftAP
+
+To enable SoftAP:
+
+1. Define and register `enabled_cb()` callback. It will be called when the SoftAP is enabled.
+   `is_requested` indicates whether this chage is requested by self or not.
+
+   ```
+   static void __enabled_cb(softap_error_e error, bool is_requested, void *data)
+   {
+     if (error != SOFTAP_ERROR_NONE) {
+       if (!is_requested)
+         return;
+       printf("Failed to enable SoftAP[0x%X]", error);
+       return;
+   }   
+
+   if (is_requested)
+     printf("SoftAP is enabled successfully");
+   else
+     printf("SoftAP is enabled by other app");
+   }
+   ```
+
+To register the callback:	
+   ```
+   ret = softap_set_enabled_cb(softap, enabled_cb, NULL);                                 
+   if (ret != SOFTAP_ERROR_NONE)
+     printf("Fail to regester the callback.");
+   ```
+
+2. Define and register `disabled_cb()` callback. It will be called when the SoftAP is disabled.
+   `code` indicates the cause of disabling.
```

Comment:
'code' means 'softap_disabled_cause_e' enumeration.

In this example:
static void __disabled_cb(softap_error_e error, softap_disabled_cause_e code, void *data)

Reply:
OK. Then can we emphasis the exact code (static void __disabled_cb(softap_error_e error, softap_disabled_cause_e code, void *data)) instead of  `code`?

Reply:
'code' is parameter's name which has 'softap_disabled_cause_e' enumeration type.
I think 'code' cannot be replaced with (static void __disabled_cb(softap_error_e error, softap_disabled_cause_e code, void data)).
Please share your opinion.

Reply:
If technically it is not fine to change, then please ignore my comments.